rllib/offline/estimators/tests/test_ope_math.py

```diff
@@ -0,0 +1,208 @@
+            delta = -delta
+        dist_inputs[torch.arange(len(inds)), old_actions] = (
+            dist_inputs[torch.arange(len(inds)), old_actions] + delta
+        ).float()
+
+        return dist_inputs, dist_class, None
+
+    def compute_log_likelihoods(
+        self,
+        actions,
+        obs_batch,
+        *args,
+        **kwargs,
+    ):
+        # used in IS and WIS
+        inds = obs_batch[:, 0]
+        old_probs = self.sample_batch[SampleBatch.ACTION_PROB][inds]
+        old_rewards = self.sample_batch[SampleBatch.REWARDS][inds]
+
+        if self.improved:
+            # assign 50 percent higher prob to those that gave a good reward and 50
+            # percent lower prob to those that gave a bad reward
+            # rewards are 1 or 2 in this case
+            new_probs = (old_rewards == 2) * 1.5 * old_probs + (
+                old_rewards == 1
+            ) * 0.5 * old_probs
+        else:
+            new_probs = (old_rewards == 2) * 0.5 * old_probs + (
+                old_rewards == 1
+            ) * 1.5 * old_probs
+
+        return np.log(new_probs)
+
+
+class TestOPEMath(unittest.TestCase):
+    """Tests some sanity checks that should pass based on the math of ope methods."""
+
+    @classmethod
+    def setUpClass(cls):
+        ray.init()
+
+        bsize = 1024
+        action_dim = 2
+        observation_space = gym.spaces.Box(-float("inf"), float("inf"), (1,))
+        action_space = gym.spaces.Discrete(action_dim)
+        cls.sample_batch = SampleBatch(
+            {
+                SampleBatch.OBS: np.arange(bsize).reshape(-1, 1),
+                SampleBatch.NEXT_OBS: np.arange(bsize).reshape(-1, 1) + 1,
+                SampleBatch.ACTIONS: np.random.randint(0, action_dim, size=bsize),
+                SampleBatch.REWARDS: np.random.randint(
+                    1, 3, size=bsize
+                ),  # rewards are 1 or 2
+                SampleBatch.DONES: np.ones(bsize),
+                SampleBatch.EPS_ID: np.arange(bsize),
+                SampleBatch.ACTION_PROB: np.ones(bsize) / action_dim,
+            }
+        )
+
+        cls.policies = {
+            "good": FakePolicy(
+                observation_space, action_space, cls.sample_batch, improved=True
+            ),
+            "bad": FakePolicy(
+                observation_space, action_space, cls.sample_batch, improved=False
+            ),
+        }
+
+    @classmethod
+    def tearDownClass(cls):
+        ray.shutdown()
+
+    def test_is_and_wis_math(self):
+        """Tests that the importance sampling and weighted importance sampling
+        methods are correct based on the math."""
+
+        ope_classes = [
+            ImportanceSampling,
+            WeightedImportanceSampling,
+        ]
+
+        for class_module in ope_classes:
+            for policy_tag in ["good", "bad"]:
+                target_policy = self.policies[policy_tag]
+                estimator_good = class_module(target_policy, gamma=0)
```

Comment:
got confused for a second. seems this should just be estimator, since it can be good and bad depends on policy_tag.

Reply:
oh good catch. misnomer :)